src/python/zquantum/core/circuit/_gate.py

```diff
@@ -377,7 +386,7 @@ def to_cirq(self, input_cirq_qubits=None):
         if self.name == "ZZ":
             return cirq.ZZ(cirq_qubits[0], cirq_qubits[1]) ** (params[0] * 2 / pi)
 
-    def to_qiskit(self, qreg=None):
+    def to_qiskit(self, qreg, creg):
```

Comment:
Shouldn't `creg` be an optional parameter that we should create as some default if it's not provided?

Reply:
I was toying with this idea, but I don't think it makes sense to create registers that are only "seen" by the Gate class. The register should be created/controlled by the circuit itself.

Reply:
Ok, sounds good to me :)